server/events/working_dir.go

```diff
@@ -117,26 +117,30 @@ func (w *FileWorkspace) Clone(
 		// We're prefix matching here because BitBucket doesn't give us the full
 		// commit, only a 12 character prefix.
 		if strings.HasPrefix(currCommit, p.HeadCommit) {
-			log.Debug("repo is at correct commit %q so will not re-clone", p.HeadCommit)
-			return cloneDir, w.warnDiverged(log, p, headRepo, cloneDir), nil
+			if w.CheckoutMerge && w.recheckDiverged(log, p, headRepo, cloneDir) {
+				log.Info("base branch has been updated, using merge strategy and will clone again")
+			} else {
+				log.Debug("repo is at correct commit %q so will not re-clone", p.HeadCommit)
+				return cloneDir, false, nil
```

Comment:
Is there a need for the boolean return if we always now return false and no longer are warning users?

Reply:
@finnag please comment

Reply:
Yes - the way the code looks now the boolean is not required. However, looking at it again, the test is probably nice to have in place to verify that we discover divergence properly, so I think reinstating the test and returning whether you had to clone or not is a better choice. Also, the test can and should be extended to test the new merging functionality. Making a fix for that now.